main/java/mjc/Runner.java

```diff
@@ -0,0 +1,184 @@
+//
+//        PurchaseAllDiscounted searchKey = new PurchaseAllDiscounted(product, 555, 0);
+//        int index = search(purchases, searchKey);
+//        if (index < 0) {
+//            System.out.println("No purchase with cost " + searchKey.getCost() + " found.");
+//        } else {
+//            System.out.println(purchases[index]);
+//        }
+
+        //Exceptions Task
+//        Comparator<AbstractPurchase> comparator = AbstractPurchase::compareTo;
+//        PurchaseList list = new PurchaseList("main/java/mjc/resources/purchases.csv", comparator);
+//        PurchaseAllDiscounted purchase = new PurchaseAllDiscounted(product, 10, 10);
+//        list.insert(-4, purchase);
+//        list.delete(4, 7);
+//        System.out.println(list.getTotalCost());
+//        System.out.println(list.toString());
+//        list.sort();
+//        System.out.println(list.binarySearch(searchKey));;
+
+        //Collections Task
+        final String FILENAME = "main/java/mjc/resources/purchases.csv";
+        Map<AbstractPurchase, Weekday> purchasesLastDayMap;
+        Map<AbstractPurchase, Weekday> purchasesFirstDayMap;
+        Map<Weekday, PurchaseList> weekdayPurchaseEnumMap;
+        PurchaseList discountedPurchases;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILENAME))) {
+            purchasesLastDayMap = new HashMap<>();
+            purchasesFirstDayMap = new HashMap<>();
+            weekdayPurchaseEnumMap = new EnumMap<>(Weekday.class);
+            discountedPurchases = new PurchaseList();
+            String day, csvLine;
+
+            while ((csvLine = reader.readLine()) != null && (day = reader.readLine()) != null) {
+                AbstractPurchase purchase = PurchaseFactory.getPurchase(csvLine);
+                Weekday weekday = Weekday.valueOf(day);
+
+                putLastDay(purchasesLastDayMap, weekday, purchase);
+                putFirstDay(purchasesFirstDayMap, weekday, purchase);
+                putEnumMapPurchase(weekdayPurchaseEnumMap, weekday, purchase);
+                if (purchase instanceof PurchaseAllDiscounted) {
+                    discountedPurchases.add(purchase);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("No source file found.");
+        } catch (CsvLineException e) {
+            throw new RuntimeException();
+        }
+
+        System.out.println("Purchase => Weekday of LAST purchase");
+        printMap(purchasesLastDayMap);
+        System.out.println("Purchase => Weekday of FIRST purchase");
+        printMap(purchasesFirstDayMap);
+
+        Purchase purchase1 = new Purchase("bread", 155, 1);
+        System.out.println("Find FIRST weekday for product " + purchase1.getProduct() + ": " + findMapValue(purchasesFirstDayMap, purchase1));
+        System.out.println("Find LAST weekday for product " + purchase1.getProduct() + ": " + findMapValue(purchasesLastDayMap, purchase1));

```

Comment:
make output inside `findMapValue` method. 
it's better to call it `findAndShowMapValue`.
don't forget to consider the case when key isn't found.